File: cypress/integration/basic.spec.js
describe('Basic tests', () => {
  beforeEach(() => {
    cy.request('/cypress_rails_reset_state');
    cy.visit('/');
  });

  it('can navigate', () => {
    cy.contains('About').click();
    cy.get('main h1').should('contain', 'About');

    cy.contains('Home').click();
    cy.get('main h1').should('contain', 'Hello');
  });

  it('can click', () => {
    cy.get('main h1').should('contain', 'Hello, Templatus!');

    cy.get('button').click();
    cy.get('ul').should('contain', '127.0.0.0');
    cy.get('li').should('have.length', 1);

    cy.get('#counter').should('contain', '1');
  });
});

File: spec/system/basic_spec.rb
describe 'Basic navigation and interaction' do
  it 'can navigate between pages' do
    visit '/'
    expect(page).to have_content('Hello')

    click_on 'About'
    expect(page).to have_selector('main h1', text: 'About')

    click_on 'Home'
    expect(page).to have_selector('main h1', text: 'Hello')
  end

  it 'can click button and see updates' do
    visit '/'
    expect(page).to have_selector('main h1', text: 'Hello, Templatus!')
    expect(page).to have_selector('#counter', text: '0')

    click_button 'Click me!'

    expect(page).to have_selector('#counter', text: '1')
    expect(page).to have_selector('ul', text: '127.0.0.0')
    expect(page).to have_selector('li', count: 1)
    expect(page).to have_selector('#flash', text: 'successfully')
  end
end
